core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -842,6 +852,18 @@ object Parser extends ParserInstances {
     }
   }
 
+  /** Parse the longest matching string between alternatives.
+    * The order of the strings does not matter.
+    *
+    * If no string matches, this parser results in an epsilon failure.
+    */
+  def stringIn1(strings: List[String]): Parser1[Unit] =
```

Comment:
since order does not matter here maybe we should take an Iterable[String] to make that clearer. List (or any Seq) tends to imply order does matter.